tools/scripts/track-pr

```diff
@@ -172,16 +172,15 @@ def cherry_pick_pr(pr_id: str) -> None:
 
     try:
         # Find and fetch the PR commit and both release branches.
+        branch_pat = r"/release-(\d+)\.(\d+)\.(\d+){}$".format("-ee" if is_ee_pr else "")
```

Comment:
Any particular reason to not just group `-ee` (non-capturing) and make that group optional?  Using a single fixed expression allows Python to cache the result of compiling the regex slightly more efficiently, which should offer incrementally better performance.  And you don't have the gross "build a regex with format" thing.
```suggestion
        branch_pat = r"/release-(\d+)\.(\d+)\.(\d+)(?:-ee)?$"
```

Reply:
(granted, this should be cached too, so that's not exactly a huge differentiatior)

Reply:
Also precision. _Probably_ we won't have a `release-...-ee` branch on OSS or vice versa, but in this case I decided it was worth defending against.

The compiled pattern caching shouldn't be affected, since the pattern is still constant per script run. I suppose having it as a literal would allow Python to use a load-constant opcode rather than a load-variable one, but I'm not about to make any decisions in the context of this script based on that level of optimization.

Though I guess I will go ahead and change it to use `re.compile` in any case.